src/feature/client/entrynodes.c

```diff
@@ -3186,13 +3273,22 @@ entry_guards_load_guards_from_state(or_state_t *state, int set)
       tor_assert(gs);
       smartlist_add(gs->sampled_entry_guards, guard);
       guard->in_selection = gs;
+      /* Recompute the next_sampled_id from the state  */
```

Comment:
Hmm, can we document when this can happen? Also why don't we do it everytime and we only do it in the `<=` case?

Reply:
Thanks for the comment!